### DISCUSSION TITLE
ERROR	Error: EROFS: read-only file system

### DISCUSSION THREAD

**Discussion by jeisonverjan** [vercel/community#1560] [Help] [answered]
### Summary

I have created an application which receives an Excel file, the information is converted into JSON format to be processed. After deploying my project to Vercel, when I try to read the file I am receiving the error message: 'Error: EROFS: read-only file system'. After investigating I have found that it is because the space that was given to me in the Vercel servers is read-only. Is there any way I can be granted read and write permissions on the path where I receive the files?

### Additional information

I'm using nodejs, express and XLSX

_No response_

### Example
This is the function:

import XLSX from 'xlsx'

export const receiveFile = (pathFile) => {
    let wb = XLSX.readFile(pathFile, {cellDates:true})
    let wbSheetsNames = wb.SheetNames
    let ws = wb.Sheets[wbSheetsNames[0]]
    

    // Modify Headers, remove spaces and make the first char lowercase
    let range = XLSX.utils.decode_range(ws['!ref'])
    for (var C = range.s.c; C <= range.e.c; ++C) {
        var address = XLSX.utils.encode_col(C) + "1"; // <-- first row, column number C
        if (!ws[address]) continue;
        ws[address].w = ws[address].w.replace(/[\(\)\s]/g, "");
        ws[address].w = ws[address].w.charAt(0).toLowerCase() + ws[address].w.slice(1)
        ws[address].v = ws[address].v.replace(/[\(\)\s]/g, "");
        ws[address].v = ws[address].v.charAt(0).toLowerCase() + ws[address].v.slice(1)
    }

    // Data converted to json format
    const wbData = XLSX.utils.sheet_to_json(ws)
    const wbHeaders = XLSX.utils.sheet_to_json(ws, {header: 1})[0]
    return [wbData, wbHeaders]
}

_No response_

**Comment by amyegan** [comment, marked as answer]
Hi Jeison! Vercel deploys projects in a serverless environment, which operates a little differently than a traditional server. Read/write permission cannot be granted because of this. But it is still possible to work with files. 

You can use the `/tmp` directory in a Serverless Function to temporarily write a file before sending or saving the file to a more permanent storage location. You can find a more detailed description of this approach in the guide on [using files in Serverless Functions on Vercel](https://vercel.com/guides/how-can-i-use-files-in-serverless-functions)
